File: Unit3E3.go
// Author: William Provost
// Version: 1.0.0
// Date: 2025-11-09
// Fileoverview: This program will ask the user to enter four lines of text.
// The program will then display the lines in reverse order (line 4, line 3, line 2, line 1).

package main

import "fmt"

func main() {
    // variables
    var line1 string
    var line2 string
    var line3 string
    var line4 string

    // input
    fmt.Print("Enter line 1: ")
    fmt.Scanln(&line1)

    fmt.Print("Enter line 2: ")
    fmt.Scanln(&line2)

    fmt.Print("Enter line 3: ")
    fmt.Scanln(&line3)

    fmt.Print("Enter line 4: ")
    fmt.Scanln(&line4)

    // output
    fmt.Println()
    fmt.Println("The lines in reverse order are:")
    fmt.Println(line4)
    fmt.Println(line3)
    fmt.Println(line2)
    fmt.Println(line1)
    fmt.Println("\nDone.")
}

File: Unit3E3.ts
/**
 * @author William Provost
 * @version 1.0.0
 * @date 2025-11-09
 * @fileoverview This program prompts the user to input four lines of text.
 * It then displays the lines in reverse order (line 4, line 3, line 2, line 1).
 */

// variables
let line1: string;
let line2: string;
let line3: string;
let line4: string;

// input
line1 = prompt("Enter line 1:") || "";
line2 = prompt("Enter line 2:") || "";
line3 = prompt("Enter line 3:") || "";
line4 = prompt("Enter line 4:") || "";

// output
console.log("\nThe lines in reverse order are:");
console.log(line4);
console.log(line3);
console.log(line2);
console.log(line1);
console.log("Done.");
